samza-sql/src/main/java/org/apache/samza/sql/data/RexToJavaCompiler.java

```diff
@@ -0,0 +1,241 @@
+
+    if (log.isDebugEnabled()) {
+      log.debug("Generated code for expression: " + s);
+    }
+
+    try {
+      return getExpression(classDeclaration, s);
+    } catch (Exception e) {
+      throw new SamzaException("Expression compilation failure.", e);
+    }
+  }
+
+  static BlockBuilder getExecuteMethodWithOnlyInputsBlockBuilder(ParameterExpression inputValues) {
+    final BlockBuilder builder = new BlockBuilder();
+    final org.apache.calcite.linq4j.tree.Expression values =
+        builder.append("values", Expressions.newArrayBounds(Object.class, 1, Expressions.constant(1)));
+    builder.add(Expressions.statement(
+        Expressions.call(Expressions.parameter(Scalar.class, "this"), SamzaBuiltInMethod.EXPR_EXECUTE2.method,
+            inputValues, values)));
+    builder.add(Expressions.return_(null, Expressions.arrayIndex(values, Expressions.constant(0))));
+
+    return builder;
+  }
+
+  static org.apache.samza.sql.data.Expression getExpression(ClassDeclaration expr, String s)
+      throws CompileException, IOException {
+    ICompilerFactory compilerFactory;
+    try {
+      compilerFactory = CompilerFactoryFactory.getDefaultCompilerFactory();
+    } catch (Exception e) {
+      throw new IllegalStateException("Unable to instantiate java compiler", e);
+    }
+    IClassBodyEvaluator cbe = compilerFactory.newClassBodyEvaluator();
+    cbe.setClassName(expr.name);
+    cbe.setImplementedInterfaces(expr.implemented.toArray(new Class[expr.implemented.size()]));
+    cbe.setParentClassLoader(RexToJavaCompiler.class.getClassLoader());
+    cbe.setDebuggingInformation(true, true, true);
+
+    //System.out.println(s);
+
+    return (org.apache.samza.sql.data.Expression) cbe.createInstance(new StringReader(s));
+  }
+
+  public enum SamzaBuiltInMethod {
+    EXPR_EXECUTE1(org.apache.samza.sql.data.Expression.class, "execute", Object[].class),
+    EXPR_EXECUTE2(org.apache.samza.sql.data.Expression.class, "execute", Object[].class, Object[].class);
+
+    public final Method method;
+    public final Constructor constructor;
+    public final Field field;
+
+    public static final ImmutableMap<Method, BuiltInMethod> MAP;
+
+    static {
+      final ImmutableMap.Builder<Method, BuiltInMethod> builder = ImmutableMap.builder();
+      for (BuiltInMethod value : BuiltInMethod.values()) {
+        if (value.method != null) {
+          builder.put(value.method, value);
+        }
+      }
+      MAP = builder.build();
+    }
+
+    private SamzaBuiltInMethod(Method method, Constructor constructor, Field field) {
+      this.method = method;
+      this.constructor = constructor;
+      this.field = field;
+    }
+
+    /**
+     * Defines a method.
+     */
+    SamzaBuiltInMethod(Class clazz, String methodName, Class... argumentTypes) {
+      this(Types.lookupMethod(clazz, methodName, argumentTypes), null, null);
+    }
+
+    /**
+     * Defines a constructor.
+     */
+    SamzaBuiltInMethod(Class clazz, Class... argumentTypes) {
```

Comment:
So, we don't seem to use the constructor and the field defining method below. Can we remove them for now until needed?